apps/ui/src/components/pages/application/form-components/ESignature.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { DownloadOutlined, EyeOutlined } from '@ant-design/icons';
+import { type eSignatureSchemaType } from '@pcgl-daco/validation';
+import { Button, Flex, Row, theme } from 'antd';
+import { type RefObject } from 'react';
+import {
+	Controller,
+	type FieldValues,
+	type FormState,
+	type UseControllerProps,
+	type UseFormClearErrors,
+	type UseFormReset,
+	type UseFormSetValue,
+	type UseFormWatch,
+} from 'react-hook-form';
+import SignatureCanvas from 'react-signature-canvas';
+
+import ErrorLabel from '@/components/pages/application/form-components/labels/ErrorLabel';
+
+interface eSignatureFormProps<T extends FieldValues> {
+	setValue: UseFormSetValue<T>;
+	formState: FormState<T>;
+	clearErrors: UseFormClearErrors<T>;
+	watch: UseFormWatch<T>;
+	reset: UseFormReset<T>;
+}
+interface eSignatureProps {
+	signatureRef: RefObject<SignatureCanvas>;
+	downloadButton: string;
+	previewButton: string;
+	clearButton: string;
+	saveButton: string;
```

Comment:
I'd change these 4 `Button` prop names to `${name}ButtonText`, they're not passing a component down they're passing a string so might as well be explicit